attrs.go

```diff
@@ -69,6 +69,19 @@ func fileInfoFromStat(stat *FileStat, name string) os.FileInfo {
 	}
 }
 
+// FileInfoUidGid extends os.FileInfo and adds callbacks for Uid and Gid retrieval.
```

Comment:
Hm, I’m not sure what the value is of exporting this interface here. Yes, it’s used immediately below, but it’s not anywhere where a caller would make use of it to understand where it is relevant?

Reply:
I've added a comment to the code to specify where it can be required. I'm using the RequestServer code to implement an sftp server that looks like a unix filesystem but is actually stored in a kind of object store. There is no reason to have *syscall.Stat_t around for just including uid/gid support.

I guess it could be an unexported interface too, of course.

Reply:
Yeah, I think our long-term plan should be that users can just return a `sshfx.Attributes` or `sshfx.NameEntry` more or less directly, meaning we can avoid needing to do too many work arounds to allow access to the full feature set.

But this is still a bit of a ways off, waiting for v2.